.github/workflows/docs.yml

```diff
@@ -13,14 +13,30 @@ jobs:
     - name: apt-update
       run: sudo apt-get update -qq
     - name: apt-get doxygen
-      run: sudo apt-get install -y doxygen
-    - name: build doc
+      run: sudo apt-get install -y doxygen graphviz
+    - name: build docs
       run: make docs
-    - name: deploy
-      uses: peaceiris/actions-gh-pages@v4
+    - name: Upload static files as artifact
+      id: deployment
+      uses: actions/upload-pages-artifact@v3 # or specific "vX.X.X" version tag for this action
```

Comment:
gh-pages requires that a branch be created specifically to hold generated artifacts. That is, the HTML site that gets generated now needs to be committed to the repo; which (IMHO) defeats one of the major purposes of automated documentation generation. 

This PR removes the dependency on a specific branch and generates the content straight from master as an artifact that github handles itself. Thus, no repo storage of an output artifact. Which can help to prevent issues where the output artifact gets out of sync with the source.

Reply:
and yes. I used the https://github.com/actions/upload-pages-artifact as a reference for this update.

Reply:
Globally I'm ok with the solution, which means no need for a specific `gh-pages` branch anymore.
But also no immediate way to see the public pages have been updated unless checking deployment jobs. Until now that commit to `gh-pages` was a clear indicator.
I'm ok with it, though.

Reply:
With the make file, we can set the standard build job to also make docs, then produce warnings or errors on documentation. That way the build or PR check fails

Reply:
Honest question. Why not enforce documentation?

In the least, it's a single line.

Reply:
Documentation building failures should not block someone from contributing to the project. But that's just like my opinion, man.